pkg/controller/nodes/task/taskexec_context.go

```diff
@@ -124,6 +143,61 @@ func (t taskExecutionContext) SecretManager() pluginCore.SecretManager {
 	return t.sm
 }
 
+// Validates and assigns a single resource by examining the default requests and max limit with the static resource value
+// defined by this task and node execution context.
+func assignResource(resourceName v1.ResourceName, execConfigRequest, execConfigLimit string, requests, limits v1.ResourceList) {
+	if len(execConfigLimit) == 0 || len(execConfigRequest) == 0 {
+		return
+	}
+	request, ok := requests[resourceName]
+	maxLimit := resource.MustParse(execConfigLimit)
```

Comment:
I would rather avoid panics if we can help it...

```suggestion
	maxLimit, err := resource.Parse(execConfigLimit)
	if err != nil {
	  return err
	}
	
	request, ok := requests[resourceName]
```

Reply:
In fact, if we change the CRD type to resource.Quantity, we don't need to do parsing here at all...